zk/tests/test_zk.py

```diff
@@ -66,3 +67,15 @@ def test_error_state(aggregator, dd_environment, get_conn_failure_config):
     expected_mode = get_conn_failure_config['expected_mode']
     mname = "zookeeper.instances.{}".format(expected_mode)
     aggregator.assert_metric(mname, value=1, count=1)
+
+
+@pytest.mark.usefixtures('dd_environment')
+def test_metadata(check, get_instance, aggregator, version_metadata):
+    check.check_id = 'test:123'
+
+    with mock.patch('datadog_checks.base.stubs.datadog_agent.set_check_metadata') as m:
+        check.check(get_instance)
+        for name, value in version_metadata.items():
+            m.assert_any_call('test:123', name, value)
+
+        assert m.call_count == len(version_metadata)
```

Comment:
A new cleaner assertion method is available, have a look at some other similar PR: https://github.com/DataDog/integrations-core/pulls?utf8=%E2%9C%93&q=is%3Apr+%22Add+version+metadata%22+in%3Atitle+

Example:

https://github.com/DataDog/integrations-core/pull/4952/files#diff-cd97faf240113603a1ac7857e7499532R40-R55

Reply:
I made the changes in this commit: https://github.com/DataDog/integrations-core/pull/4927/commits/20dbbbf3a50ccee6bd97a08ddc5a619664912b1c